contracts/modules/royalty/RoyaltyModule.sol

```diff
@@ -5,50 +5,67 @@ import { ReentrancyGuardUpgradeable } from "@openzeppelin/contracts-upgradeable/
 import { UUPSUpgradeable } from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
 import { ERC165Checker } from "@openzeppelin/contracts/utils/introspection/ERC165Checker.sol";
 import { IERC165 } from "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import { EnumerableSet } from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { BeaconProxy } from "@openzeppelin/contracts/proxy/beacon/BeaconProxy.sol";
 
 import { BaseModule } from "../BaseModule.sol";
+import { VaultController } from "./policies/VaultController.sol";
 import { IRoyaltyModule } from "../../interfaces/modules/royalty/IRoyaltyModule.sol";
 import { IRoyaltyPolicy } from "../../interfaces/modules/royalty/policies/IRoyaltyPolicy.sol";
+import { IExternalRoyaltyPolicy } from "../../interfaces/modules/royalty/policies/IExternalRoyaltyPolicy.sol";
+import { IIpRoyaltyVault } from "../../interfaces/modules/royalty/policies/IIpRoyaltyVault.sol";
 import { IDisputeModule } from "../../interfaces/modules/dispute/IDisputeModule.sol";
 import { ILicenseRegistry } from "../../interfaces/registries/ILicenseRegistry.sol";
 import { ILicensingModule } from "../../interfaces/modules/licensing/ILicensingModule.sol";
 import { Errors } from "../../lib/Errors.sol";
 import { ROYALTY_MODULE_KEY } from "../../lib/modules/Module.sol";
-import { ProtocolPausableUpgradeable } from "../../pause/ProtocolPausableUpgradeable.sol";
 
 /// @title Story Protocol Royalty Module
-/// @notice The Story Protocol royalty module allows to set royalty policies an IP asset and pay royalties as a
-///         derivative IP.
-contract RoyaltyModule is
-    IRoyaltyModule,
-    ProtocolPausableUpgradeable,
-    ReentrancyGuardUpgradeable,
-    BaseModule,
-    UUPSUpgradeable
-{
+/// @notice The Story Protocol royalty module governs the way derivatives pay royalties to their ancestors
+contract RoyaltyModule is IRoyaltyModule, VaultController, ReentrancyGuardUpgradeable, BaseModule, UUPSUpgradeable {
```

Comment:
Please also handle integration with grouping, that transfer RTs to group's pool when initialize royalty vault for the group IPA.